vars/DocMultiplicationPipeline.groovy

```diff
@@ -0,0 +1,413 @@
+def call(Map config = [:]) {
+    def params = config.params ?: [:]
+    def contexts = config.contexts ?: [:]
+    def workerAgent = config.workerAgent ?: 'Jenkins-Default-Agent-X64-C5xlarge-Single-Host'
+    
+    node(workerAgent) {
+        try {
+            echo "Starting Document Multiplication Pipeline"
+            echo "Target: Create large snapshot in S3 bucket: ${params.snapshotBucketPrefix}${params.stage}"
+            
+            env.AWS_DEFAULT_REGION = "${params.region}"
+            env.STAGE = "${params.stage}"
+            env.DEBUG_MODE = "${params.debugMode}"
```

Comment:
What is debug mode for? Is it just extra info that we should possibly normally print out?